exercises/slot_machine.livemd

```diff
@@ -0,0 +1,77 @@
+# Slot Machine
+
+## Navigation
+
+[Return Home](../start.livemd)<span style="padding: 0 30px"></span>
+[Report An Issue](https://github.com/DockYard-Academy/beta_curriculum/issues/new?assignees=&labels=&template=issue.md&title=)
+
+## The House Inc
+
+You have been hired at The House Inc.
+
+The House Inc is a casino right in the middle of Generic City. You aren't sure why a casino needs a programmer but you were desperate and applied anyway.
+
+Somehow, you actually made it past the other applicants and got the job... Or maybe that was because you were the only applicant.
+
+Your first assignment was to write the code for a slot machine. You looked at the specification and you were alarmed at how unfair this slot machine is. The House Inc isn't known for being a good place to work but you decide to talk to your boss about it anyway.
+
+So you went to your boss's office to voice your opinions on the topic and after you give a passionate speech on the value of ethics and morals. Your boss interrupts you in the middle of your ~~tantrum~~ speech and asks what ethics means.
+
+At that very moment you knew that there was no way these guys would listen because the only ethics they believe in are profit maximization.
+
+## Slot Machine Specs
+
+The slot machine is a `GenServer` that has the state of `balance` within it. Balance is how much money the machine has.
+
+* Slot machine reward sizes are random but must not exceed `balance`.
+* When a slot machine gives out a reward. Subtract `balance` by the amount of the reward.
+* The machine should only give rewards 30% of the time.
+
+This exercise will need you to use if statements. You will also need to figure out a way to generate random numbers and convert those into chances.
+
+```elixir
+defmodule SlotMachine do
+  use GenServer
+
+  def start() do
```

Comment:
Let's prefer linked processes with `start_link`. Also, to work with a supervisor these functions should always accept options, even if it's ignored.

```elixir
def start_link(_opts) do
   GenServer.start_link(__MODULE__, [])
end
```

Reply:
Let's also add an @doc with some doctests here (See below for example)